CefSharp/ModelBinding/DefaultBinder.cs

```diff
@@ -114,17 +114,14 @@ public virtual object Bind(object obj, Type modelType)
                 {
                     var val = GetValue(bindingContext, i);
 
-                    if (val != null)
+                    if (val != null && typeof(IDictionary<string, object>).IsAssignableFrom(val.GetType()))
```

Comment:
I don't think this change reads very well, think something like the following would read better, also added some extra checks.

- Assuming it's possible to have a multi dimensional array, haven't tested that yet.
- If it's an array of value types and you pass in null then use `Activator.CreateInstance` to return the default value for the type. Trying to avoid throwing exceptions as currently there is no clearly defined place to catch them.



```c#
//If the value is null then we'll add null to the collection,
if(val == null)
{
	//For value types like int we'll create the default value and assign that as we cannot assign null
	model.Add(genericType.IsValueType ? Activator.CreateInstance(genericType) : null);
}
else
{
	var valueType = val.GetType();
	//If the collection item is a list or dictionary then we'll attempt to bind it
	if (typeof(IDictionary<string, object>).IsAssignableFrom(valueType) ||
		typeof(IList<object>).IsAssignableFrom(valueType))
	{
		var subModel = Bind(val, genericType);
		model.Add(subModel);
	}
	else
	{ 
		model.Add(val);
	}
}
```

Reply:
Done, added tests to cover these scenarios